### PR TITLE
Fee manager improvements

### LINKED ISSUE #181: Dynamic fee improvements
* Per byte fee for all chains, not only OP
* Reduce constant gas limit impact
* Rename current constant fee environment variable to MIN_FEE and change the behaviour accordingly

### PR DESCRIPTION
**Pull request by lok52**
Closes #181 
This PR is a global refactoring of `FeeManager` which introduces breaking changes for client side.
Some major highlights:

1. Use different base gas limits for different tx types. These are hardcoded constants, which are still a subject for discussion. For now, they are all set to `650000`. Native convert fee is set to `200000`.
2. Change response for dynamic estimations:
```
{
  fee: {
    deposit: string;
    transfer: string;
    withdrawal: string;
    permittableDeposit: string;
  };
  oneByteFee: string;
  nativeConvertFee: string;
}
```
3. Add `RELAYER_MIN_BASE_FEE` env. Allows to set minimum fee value for base estimations (doesn't include `onyByteFee` or `nativeConvertFee`).

### CODE REVIEW

**Review by k1rill-fedoseev — CHANGES_REQUESTED** [review]

File: zp-relayer/utils/constants.ts [hunk 1]
```diff
@@ -20,6 +21,16 @@ const constants = {
   PERMIT2_CONTRACT: '0x000000000022D473030F116dDEE9F6B43aC78BA3',
   INIT_ROOT: '11469701942666298368112882412133877458305516134926649826543144744382391691533',
   MOCK_CALLDATA: '0x' + 'ff'.repeat(BASE_CALLDATA_SIZE + RAW_TX_RLP_OVERHEAD),
+  // TODO: adjust limits
+  BASE_TX_GAS: {
+    [TxType.DEPOSIT]: 650000,
```
Comment (akolotov) [inline zp-relayer/utils/constants.ts:26]: May we make these constants configured environment variables?

Comment (lok52) [reply zp-relayer/utils/constants.ts]: Moved these to env parameters

File: zp-relayer/configs/relayerConfig.ts [hunk 2]
```diff
@@ -25,6 +26,7 @@ const config = {
   relayerPrivateKey: process.env.RELAYER_ADDRESS_PRIVATE_KEY as string,
   tokenAddress: process.env.RELAYER_TOKEN_ADDRESS as string,
   relayerGasLimit: toBN(process.env.RELAYER_GAS_LIMIT as string),
+  minBaseFee: toBN(process.env.RELAYER_MIN_BASE_FEE || '0'),
```
Comment (k1rill-fedoseev) [inline zp-relayer/configs/relayerConfig.ts:29]: Let's add all new variables to CONFIGURATION.md

Comment (lok52) [reply zp-relayer/configs/relayerConfig.ts:29]: Updated

File: zp-relayer/services/price-feed/OneInchPriceFeed.ts [hunk 3]
```diff
@@ -29,29 +27,27 @@ export class OneInchPriceFeed implements IPriceFeed {
     this.contract = new web3.eth.Contract(OracleAbi as AbiItem[], contractAddress)
   }
 
+  async init() {
```
Comment (akolotov) [inline zp-relayer/services/price-feed/OneInchPriceFeed.ts:30]: Could you clarify when this code is being invoked? I tried to run the relayer with the dynamic fee manager and got the following error:
```
dev-zkbob-relayer  | /app/node_modules/number-to-bn/node_modules/bn.js/lib/bn.js:2329
dev-zkbob-relayer  |     assert(!num.isZero());
dev-zkbob-relayer  |                 ^
dev-zkbob-relayer  | 
dev-zkbob-relayer  | TypeError: Cannot read properties of undefined (reading 'isZero')
dev-zkbob-relayer  |     at BN.divmod (/app/node_modules/number-to-bn/node_modules/bn.js/lib/bn.js:2329:17)
dev-zkbob-relayer  |     at BN.div (/app/node_modules/number-to-bn/node_modules/bn.js/lib/bn.js:2425:17)
dev-zkbob-relayer  |     at OneInchPriceFeed.convert (/app/zp-relayer/services/price-feed/OneInchPriceFeed.js:42:42)
dev-zkbob-relayer  |     at /app/zp-relayer/services/fee/FeeManager.js:46:34
dev-zkbob-relayer  |     at DynamicFeeOptions.mapI (/app/zp-relayer/services/fee/FeeManager.js:24:28)
dev-zkbob-relayer  |     at DynamicFeeOptions.convert (/app/zp-relayer/services/fee/FeeManager.js:46:14)
dev-zkbob-relayer  |     at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
dev-zkbob-relayer  |     at async DynamicFeeManager.fetchFeeOptions (/app/zp-relayer/services/fee/FeeManager.js:142:31)
dev-zkbob-relayer  |     at async /app/zp-relayer/services/fee/FeeManager.js:123:32
dev-zkbob-relayer  |     at async setIntervalAndRun (/app/zp-relayer/utils/helpers.js:100:5)
dev-zkbob-relayer  |     at async DynamicFeeManager.start (/app/zp-relayer/services/fee/FeeManager.js:122:41)
dev-zkbob-relayer  |     at async init (/app/zp-relayer/init.js:92:5)
```

Comment (lok52) [reply zp-relayer/services/price-feed/OneInchPriceFeed.ts:30]: Sorry, forgot to test this after recent changes. Now, it should be ok

**Review by akolotov — APPROVED** [review]

**Pull request merged**